pkg/controllers/disruption/consolidation.go

```diff
@@ -192,6 +193,52 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 		results.NewNodeClaims[0].Requirements.Add(scheduling.NewRequirement(v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot))
 	}
 
+	results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions = instanceTypesOptionsFilteredByPrice
+	return Command{
+		candidates:   candidates,
+		replacements: results.NewNodeClaims,
+	}, nil
+}
+
+// Compute command to execute spot-to-spot consolidation if:
+// 1. The SpotToSpotConsolidation feature flag is set to true.
+// 2. There are atleast 15 cheapest instance type replacement options to consolidate.
+// 3. The current candidate is NOT part of the first 15 cheapest instance types inorder to avoid repeated consolidation for single-node consolidation.
+func (c *consolidation) computeSpotToSpotConsolidation(ctx context.Context, candidates []*Candidate,
```

Comment:
I don't think that you really need the `instanceTypeOptionsFilteredByPrice` here since you are relying on the ordered set of instance types that you are producing through this function. I may be missing it, but I don't see another reason why you need this argument here anymore

Reply:
It is changed a little now. We would still need `instanceTypeOptionsFilteredByPrice` since we would want to send only the instanceTypes that are lower priced than the current candidate for launch.